src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -187,16 +187,22 @@ public DebuggedProcess(bool bLaunched, LaunchOptions launchOptions, ISampleEngin
                         OutputMessage.Severity.Warning));
                 }
 
-                ITransport localTransport = null;
-                // For local Linux and OS X launch, use the local Unix transport which creates a new terminal and
-                // uses fifos for debugger (e.g., gdb) communication.
-                if (this.MICommandFactory.UseExternalConsoleForLocalLaunch(localLaunchOptions) &&
-                    (PlatformUtilities.IsLinux() || (PlatformUtilities.IsOSX() && localLaunchOptions.DebuggerMIMode != MIMode.Lldb)))
+                ITransport localTransport;
+
+                // Attempt to support RunInTerminal first when it is a local launch and it is not debugging a coredump.
+                // Also since we use gdb-set new-console on in windows for external console, we don't need to RunInTerminal
+                if (HostRunInTerminal.IsRunInTerminalAvailable()
+                    && String.IsNullOrWhiteSpace(localLaunchOptions.MIDebuggerServerAddress)
+                    && IsCoreDump == false
+                    && (PlatformUtilities.IsWindows() && !localLaunchOptions.UseExternalConsole))
```

Comment:
Shouldn't this last part be:
`&& (PlatformUtilities.IsWindows() ? !localLaunchOptions.UseExternalConsole : true)